src/pages/Posts.tsx

```diff
@@ -0,0 +1,112 @@
+import { useContext, useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import axios from "axios";
+
+import useLocationId from "../utils/useLocationId";
+// import useApiHook from "../utils/useApiHook";
+import { Post, PostWithComment, ViewPost } from "../interfaces";
+import { UserContentContext } from "../context/UserContentContext";
+import SinglePost from "./SinglePost";
+
+function Posts() {
+    const userId: number = useLocationId(); // Get the id parameter for URL
+    const [isLoading, setIsLoading] = useState(true); // If the posts are loading or not
+    const [viewablePost, setViewablePost] = useState<ViewPost>({
+        view: false,
+        post: {
+            post: {
+                userid: 0,
+                title: "",
+                body: "",
+                id: 0,
+            },
+            comments: [],
+        },
+        index: -1,
+    });
+
+    const { state, dispatch } = useContext(UserContentContext);
+    const { user_posts } = state;
+    useEffect(() => {
+        // If the posts in the state is of different user, then update the state,
+        // else update render with the same state
+        if (userId !== user_posts.userId) {
+            axios
```

Comment:
Extract all API calls for posts in a separate `api/posts.ts`
```
export const getUserPosts = (userId: string): Promise<PostData[]> => axios.post<AxiosResponse<PostData[]>>(...
```

And in `Posts.tsx`
```
const Posts: React.FC = () => {

    const fetchUserPosts = useCallback(async (userId: string) => {
      try {
        setLoading(true)
        const resp = await getUserPosts(userId);
        setState(resp)
      } catch (e) {
          toast('Error occured', e)
      } finally {
          setLoading(false)
      }
  }, []);

  useEffect(() => {
      fetchUserPosts(userId)
  }, [userId, fetchUserPosts])
};
```